File: backend/src/main/java/pt/isel/rugby/model/Event.java
package pt.isel.rugby.model;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import lombok.Data;

import javax.persistence.*;
import java.util.*;

@Entity
@Table(name = "event")
@Data
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private String name;

    @Column
    private String description;

    @Column
    private Date date;

    @Column
    private String local;

    @Column
   // @ManyToMany(mappedBy = "events", cascade = {CascadeType.PERSIST, CascadeType.REFRESH})
    @ManyToMany(cascade = {CascadeType.MERGE}, fetch = FetchType.LAZY)
    @JoinTable(name = "profile_events",
            joinColumns = { @JoinColumn(name = "event_id") },
            inverseJoinColumns = { @JoinColumn(name = "profile_id") })
    private List<Profile> profiles;

    public void addProfile(Profile profile){
        profiles.add(profile);
        profile.getEvents().add(this);
    }
}

File: frontend2/Rugby/src/app/classes/event.ts
import {Profile} from './profile';

export class Event {
  constructor(
    public id?: number,
    public name?: string,
    public description?: string,
    public date?: Date,
    public local?: string,
    public profiles?: any
  ) {
    this.id = id ? id : 0;
    this.description = description ? description : '';
    this.date = date ? date : new Date(0);
    this.local = local ? local : '';
    this.name = name ? name : '';
    this.profiles = profiles ? profiles : [];
  }
}
